### PR TITLE
Update common controller tests after split controller

### LINKED ISSUE #196: Improve code coverage for pkg/common-controller
Need to improve code coverage after the split controller refactor.
/assign @ggriffiths 
/cc @xing-yang 
/kind cleanup

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <grant@portworx.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Adds more tests to common controller pkg

**Which issue(s) this PR fixes**:
Part of #196 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ggriffiths** [comment]
Main tests left to cover:
- add more update snapshot class tests
- fix/update a few create snapshot tests
- update delete snapshot tests

/assign @xing-yang

**Comment by xing-yang** [comment]
Please change from "Fixes #196" to "Part of #196" so that the issue won't be closed automatically when this PR is merged.

File: pkg/common-controller/framework_test.go [hunk 1]
```diff
@@ -770,6 +761,7 @@ func newTestController(kubeClient kubernetes.Interface, clientset clientset.Inte
 	ctrl.snapshotListerSynced = alwaysReady
 	ctrl.classListerSynced = alwaysReady
 	ctrl.pvcListerSynced = alwaysReady
+	ctrl.createSnapshotContentInterval = time.Millisecond * 10
```
Comment (xing-yang) [inline pkg/common-controller/framework_test.go]: Set to 5 * time.Millisecond.

Also add a ctrl.createSnapshotContentRetryCount = 3

File: pkg/common-controller/framework_test.go [hunk 2]
```diff
@@ -1262,39 +1296,135 @@ func evaluatePVCFinalizerTests(ctrl *csiSnapshotCommonController, reactor *snaps
...
 		}
-		if test.expectSuccess && bHasPVCFinalizer {
-			t.Errorf("Test %q: failed to remove finalizer from PVC %s", test.name, test.initialClaims[0].Name)
+		bHasSnapshotFinalizer = true
+		if funcName == "testRemoveSnapshotFinalizer" {
+			for _, snapshot := range reactor.snapshots {
+				if test.initialSnapshots[0].Name == snapshot.Name {
+					if slice.ContainsString(test.initialSnapshots[0].ObjectMeta.Finalizers, utils.VolumeSnapshotBoundFinalizer, nil) && !slice.ContainsString(snapshot.ObjectMeta.Finalizers, utils.VolumeSnapshotBoundFinalizer, nil) &&
+						slice.ContainsString(test.initialSnapshots[0].ObjectMeta.Finalizers, utils.VolumeSnapshotAsSourceFinalizer, nil) && !slice.ContainsString(snapshot.ObjectMeta.Finalizers, utils.VolumeSnapshotAsSourceFinalizer, nil) {
+						klog.V(4).Infof("test %q succeeded. SnapshotFinalizer is removed from PVC %s", test.name, snapshot.Name)
```
Comment (xing-yang) [inline pkg/common-controller/framework_test.go]: s/SnapshotFinalizer is removed from PVC/SnapshotFinalizer is removed from snapshot

File: pkg/common-controller/framework_test.go [hunk 3]
```diff
@@ -1262,39 +1296,135 @@ func evaluatePVCFinalizerTests(ctrl *csiSnapshotCommonController, reactor *snaps
...
+	for _, test := range tests {
+		klog.V(4).Infof("starting test %q", test.name)
+
+		// Initialize the controller
+		kubeClient := &kubefake.Clientset{}
+		client := &fake.Clientset{}
+
+		ctrl, err := newTestController(kubeClient, client, nil, t, test)
+		if err != nil {
+			t.Fatalf("Test %q construct persistent content failed: %v", test.name, err)
```
Comment (xing-yang) [inline pkg/common-controller/framework_test.go]: s/construct persistent content failed/construct test controller failed

File: pkg/common-controller/framework_test.go [hunk 4]
```diff
@@ -1262,39 +1296,135 @@ func evaluatePVCFinalizerTests(ctrl *csiSnapshotCommonController, reactor *snaps
...
+		}
+		ctrl.classLister = storagelisters.NewVolumeSnapshotClassLister(indexer)
+
+		// Run the tested functions
+		err = test.test(ctrl, reactor, test)
+		if test.expectSuccess && err != nil {
+			t.Errorf("Test %q failed: %v", test.name, err)
+		}
+
+		// Verify Finalizer tests results
```
Comment (xing-yang) [inline pkg/common-controller/framework_test.go]: Wrong comments?

File: pkg/common-controller/snapshot_create_test.go [hunk 5]
```diff
@@ -66,230 +66,213 @@ var diffDriverStorageClass = &storage.StorageClass{
...
-			expectedSnapshots: newSnapshotArray("snap7-2", "snapuid7-2", "claim7-2", "", invalidSecretClass, "", &False, nil, nil, newVolumeError("Failed to create snapshot: failed to get input parameters to create snapshot snap7-2: \"failed to get name and namespace template from params: either name and namespace for Snapshotter secrets specified, Both must be specified\"")),
-			initialClaims:     newClaimArray("claim7-2", "pvc-uid7-2", "1Gi", "volume7-2", v1.ClaimBound, &classEmpty),
-			initialVolumes:    newVolumeArray("volume7-2", "pv-uid7-2", "pv-handle7-2", "1Gi", "pvc-uid7-2", "claim7-2", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
-			expectedEvents:    []string{"Warning SnapshotCreationFailed"},
-			errors:            noerrors,
-			test:              testSyncSnapshot,
-		},*/
+
 		{
+			name:              "7-2 - fail to update snapshot reports warning sevent",
```
Comment (xing-yang) [inline pkg/common-controller/snapshot_create_test.go]: s/warning sevent/warning event

File: pkg/common-controller/snapshot_create_test.go [hunk 6]
```diff
@@ -66,230 +66,213 @@ var diffDriverStorageClass = &storage.StorageClass{
...
-			initialVolumes:    newVolumeArray("volume7-7", "pv-uid7-7", "pv-handle7-7", "1Gi", "pvc-uid7-7", "claim7-7", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
-			expectedCreateCalls: []createCall{
+		/*
 				{
-					snapshotName: "snapshot-snapuid7-7",
-					volume:       newVolume("volume7-7", "pv-uid7-7", "pv-handle7-7", "1Gi", "pvc-uid7-7", "claim7-7", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
-					parameters:   map[string]string{"param1": "value1"},
-					// information to return
-					err: errors.New("mock create snapshot error"),
+					name:              "7-7 - fail create snapshot due to csi driver error",
```
Comment (xing-yang) [inline pkg/common-controller/snapshot_create_test.go]: This test case does not seem relevant any more as the common controller does not communicate with CSI driver directly.

**Comment by ggriffiths** [comment]
Addressed feedback @xing-yang

File: pkg/common-controller/snapshot_create_test.go [hunk 7]
```diff
@@ -66,230 +66,124 @@ var diffDriverStorageClass = &storage.StorageClass{
...
-			expectedEvents: []string{"Warning SnapshotContentCreationFailed"},
-
-			expectSuccess:     false,
-			test:           testSyncSnapshot,
-		},*/
-		/*{
-			// TODO(xiangqian): this test case needs to be revisited the scenario
-			// of VolumeSnapshotContent saving failure. Since there will be no content object
-			// in API server, it could potentially cause leaking issue
-			name:              "7-9 - fail create snapshot due to cannot save snapshot content",
```
Comment (xing-yang) [inline pkg/common-controller/snapshot_create_test.go]: 7-9 should still be kept and we'll revisit it later.

File: pkg/common-controller/snapshot_create_test.go [hunk 8]
```diff
@@ -66,230 +66,124 @@ var diffDriverStorageClass = &storage.StorageClass{
...
-					err: errors.New("mock create snapshot error"),
-				},
-			},
-			errors:         noerrors,
-			expectSuccess:     false,
-			expectedEvents: []string{"Warning SnapshotContentCreationFailed"},
-			test:           testSyncSnapshot,
-		},*/
-		/*{
-		name:              "7-8 - fail create snapshot due to cannot update snapshot status",
```
Comment (xing-yang) [inline pkg/common-controller/snapshot_create_test.go]: 7-8 should also be kept so we remember to revisit.

**Pull request merged**